From 00728ec8f4f19b5b183b40c5d628ebd1f29e02f5 Mon Sep 17 00:00:00 2001
From: Zhang Zekun <zhangzekun11@huawei.com>
Date: Tue, 16 Jul 2024 09:38:42 +0800
Subject: [PATCH 881/960] downstream: iommu/arm-smmu-v3: Change the style to
 identify the the hisi_platform

The code is isolated by CONFIG_ARCH_HISI which could also be opened
in other platforms . Change the style to identify the hisi platform
to minimize the code influence scope.

Fixes: 126526a5d9ed ("downstream: iommu/arm-smmu-v3: Enable iotlb_sync_map according to SMMU_IIDR")
Signed-off-by: Zhang Zekun <zhangzekun11@huawei.com>
Signed-off-by: Deng Guangxing <dengguangxing@huawei.com>
Signed-off-by: Zhang Guangzhi <zhangguangzhi3@huawei.com>
Signed-off-by: Li Wei <liwei818@huawei.com>
---
 drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c | 22 +++++++++++++++++++++
 1 file changed, 22 insertions(+)

diff --git a/drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c b/drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c
index 3a3ead10450e..61e59cb8333c 100644
--- a/drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c
+++ b/drivers/iommu/arm/arm-smmu-v3/arm-smmu-v3.c
@@ -139,6 +139,7 @@ DEFINE_MUTEX(arm_smmu_asid_lock);
 static struct arm_smmu_option_prop arm_smmu_options[] = {
 	{ ARM_SMMU_OPT_SKIP_PREFETCH, "hisilicon,broken-prefetch-cmd" },
 	{ ARM_SMMU_OPT_PAGE0_REGS_ONLY, "cavium,cn9900-broken-page1-regspace"},
+	{ ARM_SMMU_OPT_SYNC_MAP, "hisilicon,broken-prefetch-pgtbl"},
 	{ 0, NULL},
 };
 
@@ -5241,6 +5242,25 @@ static int arm_smmu_device_hw_probe(struct arm_smmu_device *smmu)
 }
 
 #ifdef CONFIG_ACPI
+static struct acpi_platform_list arm_smmu_v3_plat_info[] __initdata = {
+	/* HiSilicon Hip09 Platform */
+	{"HISI  ", "HIP09   ", 0, ACPI_SIG_IORT, greater_than_or_equal,
+	 "Erratum #162100602", 0},
+	{"HISI  ", "HIP10   ", 0, ACPI_SIG_IORT, greater_than_or_equal,
+	 "Erratum #162100602", 0},
+	{"HISI  ", "HIP11   ", 0, ACPI_SIG_IORT, greater_than_or_equal,
+	 "Erratum #162100602", 0},
+	{ }
+};
+
+static void acpi_get_hisi_options(struct arm_smmu_device *smmu)
+{
+	if (acpi_match_platform_list(arm_smmu_v3_plat_info) < 0)
+		return;
+
+	smmu->options |= ARM_SMMU_OPT_SYNC_MAP;
+}
+
 static void acpi_smmu_get_options(u32 model, struct arm_smmu_device *smmu)
 {
 	switch (model) {
@@ -5252,6 +5272,8 @@ static void acpi_smmu_get_options(u32 model, struct arm_smmu_device *smmu)
 		break;
 	}
 
+	acpi_get_hisi_options(smmu);
+
 	dev_notice(smmu->dev, "option mask 0x%x\n", smmu->options);
 }
 
-- 
2.43.0

